### PR TITLE
Add torchaudio with rocm 5.6 to AMD Dockerfile

### PR DESCRIPTION
**Pull request by fxmarty**
As per title

e.g. `CUDA_VISIBLE_DEVICES=0 pytest tests/models/wav2vec2/test_modeling_wav2vec2.py -s -vvvvv` now do pass.

### CODE REVIEW

**Comment by amyeroberts** [comment]
@fxmarty Could you link to any relevant conversaion / issue relating to this fix in the PR description?

**Comment by mfuntowicz** [comment]
@amyeroberts we are enabling transformers CI on AMD GPUs as part of our partnership with them.

I originally started the `huggingface:ci-amdgpu` work under a branch on the official transformers repo but @fxmarty can't do the same.

The PR here is to merge the work Felix has been doing enabling `torchaudio` to build against AMD ROCm instead of CUDA inside my original branch so we can test the workflow end2end with all the modalities.

Hope it clarify a bit what we're trying to achieve here 🙂. We can take it offline if you want some more info 👍🏼.

cc @LysandreJik for viz

**Comment by amyeroberts** [comment]
@mfuntowicz No, that's OK, thanks for the explanation! I just wanted to know if there were any open issues or additional context needed before reviews happen.

**Review by amyeroberts — APPROVED** [review]
LGTM - thanks for fixing! 

cc @ydshieh for a second opinion before we merge

File: docker/transformers-pytorch-gpu/Dockerfile.amd
Comment (ydshieh) [inline docker/transformers-pytorch-gpu/Dockerfile.amd:11]: wondering if there is any reason to move this to here?

**Review by ydshieh — COMMENTED** [review]
No special comment :-). 

~~But again, I still recommend having a working (i.e. running) version on `main` and we can start working on the failing tests.~~

Oh, sorry, it's meant to be merged into @mfuntowicz 's branch. Good!

@fxmarty can't create a branch?

File: docker/transformers-pytorch-gpu/Dockerfile.amd
Comment (fxmarty) [reply docker/transformers-pytorch-gpu/Dockerfile.amd:11]: This avoids calling `RUN` multiple times for `pip install` to avoid having too many layers in the Dockerfile, maybe it is just taste though https://docs.docker.com/develop/develop-images/dockerfile_best-practices/#minimize-the-number-of-layers

**Comment by fxmarty** [comment]
@ydshieh I think I don't have rights to create / push to a branch on transformers: https://huggingface.slack.com/archives/C014N4749J9/p1693906519779479

**Comment by mfuntowicz** [comment]
Thanks everyone for your feedback, I'm merging now to unlock the remaining on my branch 🤗

**Pull request merged**
